client/go/internal/vespa/application.go

```diff
@@ -20,7 +20,22 @@ type ApplicationPackage struct {
 	TestPath string
 }
 
-func (ap *ApplicationPackage) HasCertificate() bool { return ap.hasFile("security", "clients.pem") }
+func (ap *ApplicationPackage) HasCertificate(certPaths []string) bool {
+	if len(certPaths) == 0 {
+		if ap.hasFile("security", "clients.pem") {
+			return true
+		} else {
+			return false
+		}
+	} else {
+		for _, certPath := range certPaths {
+			if ap.hasFile(certPath) {
+				return true
+			}
+		}
+		return false
+	}
+}
 
 func (ap *ApplicationPackage) HasMatchingCertificate(certificatePEM []byte) (bool, error) {
```

Comment:
This needs to consider all certificates now, not just `clients.pem`. I think `ApplicationPackage` could have a method that reads the certificate paths from services XML, then both `HasCertificate` and `HasMatchingCertificate` can use this method and you won't have to pass around `certPaths` everywhere.

Reply:
Thanks @mpolden, I'll look at refactoring for this. The issue I had originally was that the cert.go file imports ApplicationPackage, leading to a circular import if application.go wants to make use of the file parsing from cert.go. I'll revisit and see what I can do to make it work

Reply:
Hey, I've been playing around with this and think I could do with some steer on solving it.

I've been trying to get around the circular imports by changing the `vespa` package. `xml/config.go` imports `System`, from `vespa`, but moving that to its own package ended up with a lot of files changed without solving the issue. I also tried extracting ApplicationPackage, but again couldn't get that to work on its own. There might be a way of solving it this way but I don't feel confident enough to do it as it seems it would take a more dramatic refactoring then I'm comfortable with.

The only other option I could think of was to recreate some of the xml parsing functionality within application package. It wouldn't be very dry, but would enable this. What do you think is the best way forward?

(btw I'm on the vespa slack so let me know if its easier to chat this through there)